test/functional/test_runner.py

```diff
@@ -337,6 +337,18 @@ def run_tests(test_list, src_dir, build_dir, exeext, tmpdir, jobs=1, enable_cove
     # Clear up the temp directory if all subdirectories are gone
     if not os.listdir(tmpdir):
         os.rmdir(tmpdir)
+    elif os.getenv("TRAVIS", "") == "true":
+        # Print the logs on travis, so they are preserved when the vm is disposed
+        LINES_PRINT_TRAVIS = 4000
```

Comment:
Move to .travis.yml?

Reply:
You mean, make `LINES_PRINT_TRAVIS` the environment variable instead of `TRAVIS`?

Reply:
Could it be useful to have this as a general command line option, instead of a hidden environment variable?
(no need to do in this PR, the integration just seems fairly useful)

Reply:
Yes, adding that feature to parse the max_lines in combine_logs is useful, but totally out of scope of this pull.
This code branch was only ever executed on travis and presumably by no developer. My goal here is to combine the logs on travis to debug travis specific failures.
I think placing that constant right here in the code is compact and eases code review.

Bikeshedding and new features should be submitted in separate pull request.

Reply:
My point is if in the future this value needs to be changed then a change to travis config would sound better. A nit really.